dbms/src/Storages/StorageFactory.cpp

```diff
@@ -132,6 +132,12 @@ StoragePtr StorageFactory::get(
         }
     }
 
+    if (relative_data_path.empty())
+    {
+        if (endsWith(name, "MergeTree") || endsWith(name, "Log") || name == "Join" || name == "Set")
```

Comment:
If it means "all table engines that store data on disk", I would add `Distributed`,
but better to just remove that if,
because if relative path is empty, it is logical error anyway?

Reply:
`Distributed` and `File` can be created with empty path

Reply:
Let's add a virtual method to IStorage?

Reply:
Or, better - check for nonempty path in storages constructor?